src/signer.ts

```diff
@@ -82,11 +82,7 @@ export class BlsSignerFactory {
     private constructor() {}
 
     public getSigner(domain: Domain, secretHex?: string) {
-        const secret = secretHex
-            ? secretHex.length == 66
-                ? parseFr(secretHex)
```

Comment:
After the `mcl-wasm@1.0.0` update, `parseFr` no longer works with hexes of length 64 (string length 66). It does work on hex values of length 1 (i.e. `0x1`) up to length 63, one less than what `keccak256` produces.

@ChihChengLiang Do you know why this might have changed between `mcl-wasm` versions, and if there are any implications to switching to just use `setHashFr`?

Reply:
I don't know the changes in `mcl-wasm`. Maybe #22 is related?

Reply:
I was unable to determine a way to validate that the hex is in range. So I ended up setting up the secret parsing for the signer as such:
- If the secret is missing/empty, generate a random one.
- Attempt to directly set the hex value. If this fails, fall back to hashing the secret.

I think this should cover the range of use cases and expected outputs.

Reply:
Looks like a good solution. @philsippl Would you like to provide a test case example value to make sure this change doesn't break your use case?